main.go

```diff
@@ -249,6 +248,13 @@ The log file is stored in $BITRISE_DEPLOY_DIR, and its full path is available in
 			}
 			failf("Build failed, error: %s", err)
 		}
+
+		// Cache swift PM
```

Comment:
I think caching always should be at the very end of the logic, to avoid adding anything to a cache even if the steps had a failure

Reply:
fixed.